src/clients/Deluge.ts

```diff
@@ -182,29 +213,38 @@ export default class Deluge implements TorrentClient {
 		try {
 			let torrentInfo: TorrentInfo;
 			const { duplicateCategories } = getRuntimeConfig();
+
 			if (searchee.infoHash) {
 				torrentInfo = await this.getTorrentInfo(searchee);
 				if (!torrentInfo.complete) {
-					if (
-						torrentInfo.save_path == "missing" &&
-						!path &&
-						!searchee.path
-					) {
-						return InjectionResult.FAILURE;
-					}
 					return InjectionResult.TORRENT_NOT_COMPLETE;
 				}
 			}
+			if (!path && (!searchee.infoHash || !torrentInfo)) {
+				logger.debug({
+					label: Label.DELUGE,
+					message: `Injection failure: ${newTorrent.name} was missing critical data.`,
+				});
+				return InjectionResult.FAILURE;
+			}
 
 			const params = this.formatData(
-				`${newTorrent.getFileSystemSafeName()}.cross-seed.torrent`,
+				`${newTorrent.name}.cross-seed.torrent`,
```

Comment:
what about `getFileSystemSafeName`?

Reply:
I think this PR was started before we did that, and rebasing results in it being removed as a result. My bad.